ubuntu-advantage-client/ec2_cleanup.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+"""Clean up all running EC2 instances, VPCs, and storage.
+
+Copyright 2018 Canonical Ltd.
+Joshua Powers <josh.powers@canonical.com>
+"""
+import boto3
+
+
+CI_TAG = "uaclient-*"
+
+def clean_ec2(tag):
+    """Clean up all running EC2 instances, VPCs, and storage."""
+    client = boto3.client('ec2')
+    resource = boto3.resource('ec2')
+
+    print('# searching for vpcs')
+    tag_filter = [{'Name': 'tag:Name', 'Values': [tag]}]
+    for vpc in list(resource.vpcs.filter(Filters=tag_filter)):
+        print('cleaning up vpc %s' % vpc.id)
+        for instance in vpc.instances.all():
+            print('terminating instance %s' % instance.id)
+            instance.terminate()
+            instance.wait_until_terminated()
+
+        for security_group in vpc.security_groups.filter(Filters=tag_filter):
+            print('terminating security group %s' % security_group.id)
+            security_group.delete()
+
+        for subnet in vpc.subnets.filter(Filters=tag_filter):
+            print('terminating subnet %s' % subnet.id)
+            subnet.delete()
+
+        for route_table in vpc.route_tables.filter(Filters=tag_filter):
+            print('terminating route table %s' % route_table.id)
+            route_table.delete()
+
+        for internet_gateway in vpc.internet_gateways.filter(Filters=tag_filter):
+            print('terminating internet gateway %s' % internet_gateway.id)
+            internet_gateway.detach_from_vpc(VpcId=vpc.id)
+            internet_gateway.delete()
+
+        print('terminating vpc %s' % vpc.id)
+        vpc.delete()
+
+    print('# searching for ssh keys')
+    key_name = {'Name': 'key-name', 'Values': [CI_TAG]}
```

Comment:
I think we should the `tag` parameter here instead

Reply:
ahh yes I just saw that too. Will fix now